extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    /**
+     * Handles all the operations that can be recorded in the RUNTIME_INIT execution time phase
+     * 
+     * @param template Used to set the runtime config
+     * @param agroalRuntimeConfig The Agroal runtime config to get the datasource fields
+     * @param flywayRuntimeConfig The Flyway configuration
+     */
+    @Record(ExecutionTime.RUNTIME_INIT)
+    @BuildStep
+    void configureRuntimeProperties(FlywayTemplate template,
+            AgroalRuntimeConfig agroalRuntimeConfig,
+            FlywayRuntimeConfig flywayRuntimeConfig) {
+        template.configureAgroalProperties(agroalRuntimeConfig);
+        template.configureFlywayProperties(flywayRuntimeConfig);
+    }
+
+    private void registerSubstrateResources(BuildProducer<SubstrateResourceBuildItem> resource,
+            BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer,
+            FlywayBuildConfig flywayBuildConfig)
+            throws IOException, URISyntaxException {
+        List<String> nativeResources = generateDatabasesSQLFiles();
+        List<String> applicationMigrations = discoverApplicationMigrations(flywayBuildConfig);
+        nativeResources.addAll(applicationMigrations);
+        // Store application migration in a generated resource that will be accessed later by the Quarkus-Flyway path scanner
+        String resourcesList = applicationMigrations
+                .stream()
+                .collect(Collectors.joining("\n", "", "\n"));
+        generatedResourceProducer.produce(
+                new GeneratedResourceBuildItem(
+                        QuarkusPathLocationScanner.MIGRATIONS_LIST_FILE,
+                        resourcesList.getBytes(StandardCharsets.UTF_8)));
+        nativeResources.add(QuarkusPathLocationScanner.MIGRATIONS_LIST_FILE);
+        resource.produce(new SubstrateResourceBuildItem(nativeResources.toArray(new String[0])));
+    }
+
+    private List<String> discoverApplicationMigrations(FlywayBuildConfig flywayBuildConfig)
+            throws IOException, URISyntaxException {
+        List<String> resources = new ArrayList<>();
+        try {
+            List<String> locations = flywayBuildConfig.locations;
+            if (locations.isEmpty()) {
+                locations.add("db/migration");
+            }
+            // Locations can be a comma separated list
+            for (String location : locations) {
+                Enumeration<URL> migrations = Thread.currentThread().getContextClassLoader().getResources(location);
+                while (migrations.hasMoreElements()) {
+                    URL path = migrations.nextElement();
+                    LOGGER.info("Adding application migrations in path: " + path);
+                    Set<String> applicationMigrations = walk(Paths.get(path.toURI()))
+                            .filter(Files::isRegularFile)
+                            .map(it -> Paths.get(location, it.getFileName().toString()).toString())
```

Comment:
I am wondering if this line could just be `.map(it -> it.toAbsolutePath().toString())`.

Reply:
I changed it but it didn't work. This stream mapping is generating the classpath location in the form it will be included in the native image. The migration files are shipped within the image and the `.toAbsolutePath()` method returns a physical path that won't be available at runtime.

```
2019-04-22 07:20:08,327 DEBUG [org.fly.cor.Flyway] (executor-thread-1) Memory usage: 13 of 6553M
2019-04-22 07:20:08,327 ERROR [io.und.request] (executor-thread-1) UT005023: Exception handling request to /flyway/migrate: org.jboss.resteasy.spi.UnhandledException: org.flywaydb.core.api.FlywayException: Unable to obtain inputstream for resource: /Users/cristhiank/Dev/contrib/quarkus/integration-tests/flyway/target/classes/db/location1/V1.0.1__Quarkus.sql
        at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:209)

```

Reply:
These are the transformation results:
```
[INFO] [io.quarkus.flyway.FlywayProcessor] Adding application migrations in path: file:/Users/cristhiank/Dev/contrib/quarkus/integration-tests/flyway/target/classes/db/location1
[INFO] [io.quarkus.flyway.FlywayProcessor] Discovered Before: /Users/cristhiank/Dev/contrib/quarkus/integration-tests/flyway/target/classes/db/location1/V1.0.1__Quarkus.sql
[INFO] [io.quarkus.flyway.FlywayProcessor] Discovered After: db/location1/V1.0.1__Quarkus.sql
[INFO] [io.quarkus.flyway.FlywayProcessor] Adding application migrations in path: file:/Users/cristhiank/Dev/contrib/quarkus/integration-tests/flyway/target/classes/db/location2
[INFO] [io.quarkus.flyway.FlywayProcessor] Discovered Before: /Users/cristhiank/Dev/contrib/quarkus/integration-tests/flyway/target/classes/db/location2/V1.0.0__Quarkus.sql
[INFO] [io.quarkus.flyway.FlywayProcessor] Discovered After: db/location2/V1.0.0__Quarkus.sql
```
It maps from a physical absolute path to a relative classpath accesible location

Reply:
Ah sure, thanks for explanation.